### ISSUE TITLE
Index out of Range, with 1.6.0-beta

### ISSUE THREAD

**Issue by Jacko1394** [jamesmontemagno/mvvm-helpers#57]
Hi James, thanks for this Awesome little library!

Since #56 #nullable enable, 1.6.0-beta:

When doing AddRange to a new, empty ObservableRangeCollection, I get an Index out of range exception.

Many thanks,

**Comment by Jacko1394** [comment]
The error seems to be happening in Xamarin's ListViewRenderer.
Possible relation to #50

**Comment by Sharaf0** [comment]
Hey @Jacko1394, Could you please provide code snippet with the error provided?

**Comment by Jacko1394** [comment]
```csharp
		private async Task Init() {

			IsBusy = true;
![Screen Shot 2019-11-11 at 5 32 29 pm](https://user-images.githubusercontent.com/13980469/68566080-7d83df80-04a9-11ea-9daa-50c031364e37.png)


			var init = Menu.Select(x => new Grouping<string, NavigationItem>(x.Key, x.Value)).ToArray();

			IEnumerable<EnterpriseMenuItem> list = new List<EnterpriseMenuItem>();

			try {
				list = await _enterpriseClient.GetMenuItems();
			} catch (Exception ex) {
				_logger.LogError($"MENU: {ex.Message}", ex);
			}

			var navItems = list.OrderBy(x => x.Text).Select(x => new NavigationItem(x.Text, null, x.Action));
			init[0].AddRange(navItems);

			if (list.Select(x => x.Text).Contains("Documents")) {

				_ = App.ReloadInfoRouter();

				var upload = new NavigationItem(Constants.Upload, typeof(UploadPage));
				var offline = new NavigationItem(Constants.Offline, typeof(OfflineDocsPage));

				init[1].Insert(0, upload);
				init[1].Insert(0, offline);
			}


			Groupings.Clear();
			IsBusy = false;
			Groupings.AddRange(init);


		}
```

**Comment by Jacko1394** [comment]
```csharp
		private async Task Init() {

			IsBusy = true;

			var init = Menu.Select(x => new Grouping<string, NavigationItem>(x.Key, x.Value)).ToArray();

			IEnumerable<EnterpriseMenuItem> list = new List<EnterpriseMenuItem>();

			try {
				list = await _enterpriseClient.GetMenuItems();
			} catch (Exception ex) {
				_logger.LogError($"MENU: {ex.Message}", ex);
			}

			var navItems = list.OrderBy(x => x.Text).Select(x => new NavigationItem(x.Text, null, x.Action));
			init[0].AddRange(navItems);

			if (list.Select(x => x.Text).Contains("Documents")) {

				_ = App.ReloadInfoRouter();

				var upload = new NavigationItem(Constants.Upload, typeof(UploadPage));
				var offline = new NavigationItem(Constants.Offline, typeof(OfflineDocsPage));

				init[1].Insert(0, upload);
				init[1].Insert(0, offline);
			}


			Groupings.Clear();
			IsBusy = false;
			Groupings.AddRange(init);


		}
```
![Screen Shot 2019-11-11 at 5 32 29 pm](https://user-images.githubusercontent.com/13980469/68566129-9ab8ae00-04a9-11ea-813c-2036f3e439a2.png)

**Comment by Jacko1394** [comment]
the Clear()... then AddRange was my work-around for #38 
which is fixed now, but now this crashes haha

**Comment by jamesmontemagno** [comment]
Can you give me a stack trace of the crash?

**Comment by jamesmontemagno** [comment]
Also does 1.5.X-beta work?

**Comment by jamesmontemagno** [comment]
I think this is fixed now in 1.6.1

**Comment by AbdellahOusrhireDev** [comment]
@jamesmontemagno Hi!

I'm getting the same exception on ios(simulator and iphone device); Android works fine!

Nugets:
   <PackageReference Include="Xamarin.Forms" Version="4.8.0.1187-pre2" />
   <PackageReference Include="Refractored.MvvmHelpers" Version="1.6.2" />

Trace:

```
    at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:161
    at System.Collections.ObjectModel.Collection`1[T].System.Collections.IList.get_Item (System.Int32 index) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/external/corefx/src/Common/src/CoreLib/System/Collections/ObjectModel/Collection.cs:266
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.ElementAt (System.Int32 index) [0x0000f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:254
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.get_Item (System.Int32 index) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:40
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.get_Item (Foundation.NSIndexPath indexPath) [0x00019] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:96
    at Xamarin.Forms.Platform.iOS.ItemsViewController`1[TItemsView].UpdateTemplatedCell (Xamarin.Forms.Platform.iOS.TemplatedCell cell, Foundation.NSIndexPath indexPath) [0x00012] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewController.cs:199
    at Xamarin.Forms.Platform.iOS.ItemsViewController`1[TItemsView].GetCell (UIKit.UICollectionView collectionView, Foundation.NSIndexPath indexPath) [0x00033] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewController.cs:84
    at Xamarin.Forms.Platform.iOS.ItemsViewController`1[TItemsView].GetPrototype () [0x0005d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewController.cs:263
    at Xamarin.Forms.Platform.iOS.ItemsViewLayout.DetermineCellSize () [0x0004e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewLayout.cs:201
    at Xamarin.Forms.Platform.iOS.ListViewLayout.ConstrainTo (CoreGraphics.CGSize size) [0x0001e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ListViewLayout.cs:16
    at Xamarin.Forms.Platform.iOS.ItemsViewLayout.UpdateConstraints (CoreGraphics.CGSize size) [0x0003f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewLayout.cs:96
    at Xamarin.Forms.Platform.iOS.ItemsViewLayout.InvalidateLayout () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ItemsViewLayout.cs:535
    at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend(intptr,intptr)
    at UIKit.UICollectionView.ReloadData () [0x0000d] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.18.3.2/src/Xamarin.iOS/UICollectionView.g.cs:687
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.Reload () [0x00071] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:142
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.CollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs args) [0x00198] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:129
    at Xamarin.Forms.Platform.iOS.ObservableItemsSource.CollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs args) [0x000b7] in D:\a\1\s\Xamarin.Forms.Platform.iOS\CollectionView\ObservableItemsSource.cs:108
    at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021
    at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.18.3.2/src/Xamarin.iOS/Foundation/NSAction.cs:178 --- End of stack trace from previous location where exception was thrown ---
    at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
    at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.18.3.2/src/Xamarin.iOS/UIKit/UIApplication.cs:86
    at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.18.3.2/src/Xamarin.iOS/UIKit/UIApplication.cs:65
    at Mobile.iOS.Application.Main (System.String[] args) [0x00001] in /Users/abdellahousrhire/Desktop/devinnov/Wiqaytna/Mobile/Mobile.iOS/Main.cs:17 
```
